README.md
# Rails Error Generator

[![Builds](https://github.com/chiefpansancolt/rails-error-generator/workflows/Builds/badge.svg)]()
[![Lints](https://github.com/chiefpansancolt/rails-error-generator/workflows/Linter/badge.svg)]()
[![Tests](https://github.com/chiefpansancolt/rails-error-generator/workflows/Tests/badge.svg)]()
![GitHub release](https://img.shields.io/github/release/chiefpansancolt/rails-error-generator?logo=github&style=flat-square)
![Gem](https://img.shields.io/gem/dt/rails-error-generator?logo=rubygems&style=flat-square)
[![Discord](https://img.shields.io/discord/450095227185659905?label=Discord&logo=discord&style=flat-square)](https://discord.gg/FPfA3w6)

Rails Error Generator was built to enable easy start for using Rails route rendering to render error codes using ERB over public folder. This tool will generate various version of views.

## Supported Error Codes

### 4xx Client Codes

| Code | Meaning                         | Routing                          |
| ---  | ---                             | ---                              |
| 400  | Bad Request                     | :bad_request                     |
| 401  | Unauthorized                    | :unauthorized                    |
| 402  | Payment Required                | :payment_required                |
| 403  | Forbidden                       | :forbidden                       |
| 404  | Not Found                       | :not_found                       |
| 405  | Method Not Allowed              | :method_not_allowed              |
| 406  | Not Acceptable                  | :not_acceptable                  |
| 407  | Proxy Authentication Required   | :proxy_authentication_required   |
| 408  | Request Timeout                 | :request_timeout                 |
| 409  | Conflict                        | :conflict                        |
| 410  | Gone                            | :gone                            |
| 411  | Length Required                 | :length_required                 |
| 412  | Precondition Failed             | :precondition_failed             |
| 413  | Request Entity Too Large        | :request_entity_too_large        |
| 414  | Request-URI Too Long            | :request_uri_too_long            |
| 415  | Unsupported Media Type          | :unsupported_media_type          |
| 416  | Requested Range Not Satisfiable | :requested_range_not_satisfiable |
| 417  | Expectation Failed              | :expectation_failed              |
| 422  | Unprocessable Entity            | :unprocessable_entity            |
| 423  | Locked                          | :locked                          |
| 424  | Failed Dependency               | :failed_dependency               |
| 426  | Upgrade Required                | :upgrade_required                |

### 5xx Server Codes

| Code | Meaning                         | Routing                          |
| ---  | ---                             | ---                              |
| 500  | Internal Server Error           | :internal_server_error           |
| 501  | Not Implemented                 | :not_implemented                 |
| 502  | Bad Gateway                     | :bad_gateway                     |
| 503  | Service Unavailable             | :service_unavailable             |
| 504  | Gateway Timeout                 | :gateway_timeout                 |
| 505  | HTTP Version Not Supported      | :http_version_not_supported      |
| 507  | Insufficient Storage            | :insufficient_storage            |
| 510  | Not Extended                    | :not_extended                    |

## Supported View Forms

The below list is a list of supported view formatters available for generation to match project usage.

- HTML (ERB)
- HAML
- SLIM

## Installation

TODO: Installation instructions go here

## Usage

TODO: Write usage instructions here

## Development

After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake test` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.

To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, the repo owner will publish and determine new version number prior to publishing to github and rubygems.

## Contributing

Bug reports and pull requests are welcome on GitHub at https://github.com/chiefpansancolt/rails-error-generator. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [Contributor Covenant](http://contributor-covenant.org) code of conduct.

## License

The gem is available as open source under the terms of the [MIT License](https://opensource.org/licenses/MIT).

## Code of Conduct

Everyone interacting in the Rails::Error::Generator project’s codebases, issue trackers, chat rooms and mailing lists is expected to follow the [code of conduct](https://github.com/chiefpansancolt/rails-error-generator/blob/master/CODE_OF_CONDUCT.md).


CHANGELOG.md
# Changelog

## Version [0.1.0](https://github.com/chiefpansancolt/rails-error-generator/releases/tag/0.1.0)

### Feature Enhancements
- N/A

### Bugs
- N/A

### Breaking Changes
- N/A

test/test_helper.rb
# frozen_string_literal: true

$LOAD_PATH.unshift File.expand_path("../lib", __dir__)
require "rails/error/generator"

require "minitest/autorun"


lib/generators/templates/controllers/errors.rb
class ErrorsController < ApplicationController

  # 400 Client Code
  def bad_request
    render status: :bad_request
  end

  # 401 Client Code
  def unauthorized
    render status: :unauthorized
  end

  # 402 Client Code
  def payment_required
    render status: :payment_required
  end

  # 403 Client Code
  def forbidden
    render status: :forbidden
  end
  
  # 404 Client Code
  def not_found
    render status: :not_found
  end

  # 405 Client Code
  def method_not_allowed
    render status: :method_not_allowed
  end

  # 406 Client Code
  def not_acceptable
    render status: :not_acceptable
  end

  # 407 Client Code
  def proxy_authentication_required
    render status: :proxy_authentication_required
  end

  # 408 Client Code
  def request_timeout
    render status: :request_timeout
  end
  
  # 409 Client Code
  def conflict
    render status: :conflict
  end

  # 410 Client Code
  def gone
    render status: :gone
  end

  # 411 Client Code
  def length_required
    render status: :length_required
  end

  # 412 Client Code
  def precondition_failed
    render status: :precondition_failed
  end

  # 413 Client Code
  def request_entity_too_large
    render status: :request_entity_too_large
  end

  # 414 Client Code
  def request_uri_too_long
    render status: :request_uri_too_long
  end

  # 415 Client Code
  def unsupported_media_type
    render status: :unsupported_media_type
  end

  # 416 Client Code
  def requested_range_not_satisfiable
    render status: :requested_range_not_satisfiable
  end

  # 417 Client Code
  def expectation_failed
    render status: :expectation_failed
  end

  # 422 Client Code
  def unprocessable_entity
    render status: :unprocessable_entity
  end

  # 423 Client Code
  def locked
    render status: :locked
  end

  # 424 Client Code
  def failed_dependency
    render status: :failed_dependency
  end

  # 426 Client Code
  def upgrade_required
    render status: :upgrade_required
  end

  # 500 Server Code
  def internal_server_error
    render status: :internal_server_error
  end

  # 501 Server Code
  def not_implemented
    render status: :not_implemented
  end

  # 502 Server Code
  def bad_gateway
    render status: :bad_gateway
  end

  # 503 Server Code
  def service_unavailable
    render status: :service_unavailable
  end

  # 504 Server Code
  def gateway_timeout
    render status: :gateway_timeout
  end

  # 505 Server Code
  def http_version_not_supported
    render status: :http_version_not_supported
  end

  # 507 Server Code
  def insufficient_storage
    render status: :insufficient_storage
  end

  # 510 Server Code
  def not_extended
    render status: :not_extended
  end
end